db/migrate/20201124155809_enforce_one_historic_report_per_series.rb

```diff
@@ -0,0 +1,10 @@
+class EnforceOneHistoricReportPerSeries < ActiveRecord::Migration[6.0]
+  def change
+    change_table :reports do |t|
+      t.index [:fund_id, :organisation_id],
+        where: "financial_quarter IS NULL OR financial_year IS NULL",
```

Comment:
Partial indexes are new to me 🤔

Reply:
I only just learned about them but they're really neat for being able to scope uniqueness constraints.